### PR TITLE
Add possiblity for delayed reactivation of blocking

### LINKED ISSUE #880: Time parameter for enable/disable command [Feature Request]
**In raising this issue, I confirm the following (please check boxes, eg [X]) Failure to fill the template will close your issue:**
- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [X] The issue I am reporting can be _replicated_
- [X] The issue I'm reporting isn't a duplicate (see [FAQs](https://github.com/pi-hole/pi-hole/wiki/FAQs), [closed issues](https://github.com/pi-hole/pi-hole/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20), and [open issues](https://github.com/pi-hole/pi-hole/issues)).

**How familiar are you with the codebase?:**
- [X] 1 (very unfamiliar)
- [] 2
- [] 3
- [] 4
- [] 5
- [] 6
- [] 7
- [] 8
- [] 9
- [] 10 (very familiar)

---

**[FEATURE REQUEST]:**
**One feature I really enjoy about the enable/disable script found on this script: https://www.reddit.com/r/pihole/comments/49aoqv/temporarily_stop_pihole/
Is the ability to set a time parameter, that Pihole will automatically re-enable after
For example, instead of saying "pihole disable", I would like the option of doing "pihole disable 5m" which will disable pihole for 5 minutes, after which point it will automatically be re-enabled**

_This template was created based on the work of [`udemy-dl`](https://github.com/nishad/udemy-dl/blob/master/LICENSE)._

### PR DESCRIPTION
**Pull request by DL6ER**
**By submitting this pull request, I confirm the following (please check boxes, eg [X])
Failure to fill the template will close your PR:**

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [X] I have checked that [another pull request](https://github.com/pi-hole/pi-hole/pulls) for this purpose does not exist.
- [X] I have considered, and confirmed that this submission will be valuable to others.
- [X] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [X] I give this submission freely, and claim no ownership to its content.

**How familiar are you with the codebase?:**

- [ ] 1 (very unfamiliar)
- [ ] 2
- [ ] 3
- [ ] 4
- [ ] 5
- [ ] 6
- [ ] 7
- [X] 8
- [ ] 9
- [ ] 10 (very familiar)

---

Fixes #880 

Adds possibility to disable blocking only temporarily.
Currently, it is possible to specify the delay in seconds and minutes:

- `pihole disable 5s` will disable blocking for 5 seconds
- `pihole disable 7m` will disable blocking for 7 minutes
- etc.

Note that this will even work if the user logs out from the remote connection in the meantime.

_This template was created based on the work of [`udemy-dl`](https://github.com/nishad/udemy-dl/blob/master/LICENSE)._

### CODE REVIEW

File: pihole [hunk 1]
```diff
@@ -117,6 +117,32 @@ piholeEnable() {
...
+          nohup bash -c "sleep ${tt}; pihole enable" </dev/null &>/dev/null &
+        else
+          echo "::: Unknown format for delayed reactivation of the blocking!"
+          echo "::: Example:"
+          echo ":::    pihole disable 5s    -    will disable blocking for 5 seconds"
+          echo ":::    pihole disable 7m    -    will disable blocking for 7 minutes"
+          echo "::: Blocking will not automatically be reenabled!"
+        fi
+      else
+        echo "::: Command 'nohup' has to be available for this feature."
```
Comment (PromoFaux) [inline pihole]: We could always add it to the dependency list...

Comment (DL6ER) [inline pihole]: Quickly verified that is should always be available. Only some linux based routers might not have it. So we should be on the safe side here.

Comment (DL6ER) [inline pihole]: Just found on reddit a comment 
> I ran in to a situation where I was working on a router that for some weird reason did not have the nohup command.

Don't know how credible that is, though.

File: pihole [hunk 2]
```diff
@@ -117,6 +117,24 @@ piholeEnable() {
...
+        tt=${2%"m"}
+        echo "::: Blocking will be reenabled in ${tt} minutes"
+        tt=$((${tt}*60))
+        nohup bash -c "sleep ${tt}; pihole enable" </dev/null &>/dev/null &
+      else
+        echo "::: Unknown format for delayed reactivation of the blocking!"
+        echo "::: Example:"
+        echo ":::    pihole disable 5s    -    will disable blocking for 5 seconds"
+        echo ":::    pihole disable 7m    -    will disable blocking for 7 minutes"
+        echo "::: Blocking will not automatically be reenabled!"
```
Comment (Mcat12) [inline pihole]: reenabled -> re-enabled

Comment (DL6ER) [inline pihole]: fixed

Comment (Mcat12) [inline pihole]: 2 more spots, thanks.

File: pihole (hunk 1 above)
Comment (dschaper) [inline pihole]: Well, I do appreciate the research you're doing on your commits.

File: pihole (hunk 2 above)
Comment (DL6ER) [inline pihole]: @Mcat12 done.

**Comment by Mcat12** [comment]
Going to test soon

**Comment by Mcat12** [comment]
Can you add some text to the main `pihole` help that indicates that you can set a time delay for disabling and the format? Otherwise, it works fine.

**Comment by DL6ER** [comment]
```
pi@raspberrypi:~# pihole --help
::: Control all PiHole specific functions!
:::
::: Usage: pihole [options]
:::             Add -h after -w (whitelist), -b (blacklist), or -c (chronometer)  for more information on usage
:::
::: Options:
:::  -w, whitelist            Whitelist domains
:::  -b, blacklist            Blacklist domains
:::  -d, debug                Start a debugging session if having trouble
:::  -f, flush                Flush the pihole.log file
:::  -up, updatePihole        Update Pi-hole
:::  -g, updateGravity        Update the list of ad-serving domains
:::  -s, setupLCD             Automatically configures the Pi to use the 2.8 LCD screen to display stats on it
:::  -c, chronometer          Calculates stats and displays to an LCD
:::  -h, help                 Show this help dialog
:::  -v, version              Show current versions
:::  -q, query                Query the adlists for a specific domain
:::  -l, logging              Enable or Disable logging (pass 'on' or 'off')
:::  uninstall                Uninstall Pi-Hole from your system :(!
:::  status                   Is Pi-Hole Enabled or Disabled
:::  enable                   Enable Pi-Hole DNS Blocking
:::  disable                  Disable Pi-Hole DNS Blocking
:::                           Blocking can also be disabled only temporarily, e.g.,
:::                           pihole disable 5m - will disable blocking for 5 minutes
:::  restartdns               Restart dnsmasq
```

**Comment by PromoFaux** [comment]
Approved

**Pull request merged**